From e6b1c288b2568f6c0e87deac07be07729bd853c2 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Sun, 31 Oct 2021 14:44:45 -0500
Subject: [PATCH] support pypy which doesn't have a makefile

---
 crossenv/__init__.py | 24 +++++++++++-------------
 1 file changed, 11 insertions(+), 13 deletions(-)

diff --git a/crossenv/__init__.py b/crossenv/__init__.py
index d84cb27..f52aaf1 100644
--- a/crossenv/__init__.py
+++ b/crossenv/__init__.py
@@ -322,20 +322,18 @@ def find_host_python(self, host):
         self.host_version = self.host_sysconfigdata.build_time_vars['VERSION']
         self.host_gnu_type = self.host_sysconfigdata.build_time_vars['HOST_GNU_TYPE']
 
-        # Ask the makefile a few questions too
-        if not os.path.exists(self.host_makefile):
-            raise FileNotFoundError("Cannot find Makefile")
-
         self.host_platform = None
-        with open(self.host_makefile, 'r') as fp:
-            lines = list(fp.readlines())
-        for line in lines:
-            line = line.strip()
-            if line.startswith('_PYTHON_HOST_PLATFORM='):
-                host_platform = line.split('=',1)[-1].strip()
-                if host_platform:
-                    self.host_platform = host_platform
-                break
+        # Ask the makefile a few questions too
+        if os.path.exists(self.host_makefile):
+            with open(self.host_makefile, 'r') as fp:
+                lines = list(fp.readlines())
+            for line in lines:
+                line = line.strip()
+                if line.startswith('_PYTHON_HOST_PLATFORM='):
+                    host_platform = line.split('=',1)[-1].strip()
+                    if host_platform:
+                        self.host_platform = host_platform
+                    break
 
         if self.host_platform is None:
             # It was probably natively compiled, but not necessarily for this
