From 0aa3cc9c04efc04a6b2833861ad28c8e154347b9 Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Sat, 22 Aug 2020 11:27:57 -0500
Subject: [PATCH 2/2] Fix sys.platform in cross-python

Fixes #28
---
 crossenv/scripts/site.py.tmpl | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/crossenv/scripts/site.py.tmpl b/crossenv/scripts/site.py.tmpl
index e82a5ca..4898861 100644
--- a/crossenv/scripts/site.py.tmpl
+++ b/crossenv/scripts/site.py.tmpl
@@ -148,6 +148,8 @@ def uname2():
     return uname_result2
 platform.uname = uname2
 
+sys.platform = uname_result2.system.lower()
+
 if hasattr(platform, '_linux_distribution'):
     @wraps(platform._linux_distribution)
     def dist(*args, **kwargs):
-- 
2.28.0

