From 2a823d76f47d3bf73af045b4873cdc593225adf3 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Thu, 7 Oct 2021 10:30:05 +0200
Subject: [PATCH 3/3] Ignore some values in host_sysconfigdata

---
 crossenv/__init__.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/crossenv/__init__.py b/crossenv/__init__.py
index c7f1b74..274f1bd 100644
--- a/crossenv/__init__.py
+++ b/crossenv/__init__.py
@@ -737,6 +737,8 @@ class CrossEnvBuilder(venv.EnvBuilder):
         for key, value in self.host_sysconfigdata.__dict__.items():
             if key.startswith('__'):
                 continue # misc module stuff like __name__, __builtins__
+            if key in ["_imp", "os", "sys", "find_executable"]:
+                continue
             cross_sysconfig_data[key] = value
 
         build_time_vars = {}
-- 
2.33.0

