From 65ad2066e88073b62a34d2bf55b9928187095bd0 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Wed, 29 Dec 2021 20:50:18 +0100
Subject: [PATCH] Provide empty lines if there is no Makefile

---
 crossenv/__init__.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/crossenv/__init__.py b/crossenv/__init__.py
index f52aaf1..7859347 100644
--- a/crossenv/__init__.py
+++ b/crossenv/__init__.py
@@ -334,6 +334,8 @@ class CrossEnvBuilder(venv.EnvBuilder):
                     if host_platform:
                         self.host_platform = host_platform
                     break
+        else:
+            lines = []
 
         if self.host_platform is None:
             # It was probably natively compiled, but not necessarily for this
-- 
2.34.1

