From 5fa051948d551b98f8a5e1e96e811459251165a0 Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Thu, 8 Oct 2020 17:31:46 -0500
Subject: [PATCH] support python 3.9

---
 crossenv/scripts/site.py.tmpl | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/crossenv/scripts/site.py.tmpl b/crossenv/scripts/site.py.tmpl
index 47f7e5e..34c92b2 100644
--- a/crossenv/scripts/site.py.tmpl
+++ b/crossenv/scripts/site.py.tmpl
@@ -31,13 +31,11 @@ os.environ['PYTHON_CROSSENV'] = 'x'
 # We'll try not to do that unless we really have to.
 for name in ['_PYTHON_PROJECT_BASE', '_PYTHON_HOST_PLATFORM',
              '_PYTHON_SYSCONFIGDATA_NAME', 'PYTHONHOME', 'PYTHONPATH']:
-    # This is the same as os.putenv/os.unsetenv, but I want to be clear that
-    # I'm trying to break something.
     try:
-        os.environ.putenv(name, os.environ['_OLD_' + name])
+        os.putenv(name, os.environ['_OLD_' + name])
         del os.environ['_OLD_' + name]
     except KeyError:
-        os.environ.unsetenv(name)
+        os.unsetenv(name)
 
 
 # sysconfig isn't _quite_ set up right, because it queries from a not-yet fixed
