From 2f38528417edf7e524aec4f5907b77d5ddf87722 Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Sun, 23 Jan 2022 17:45:22 -0600
Subject: [PATCH] Support patching distutils.sysconfig with pypy

---
 crossenv/scripts/site.py.tmpl | 1 +
 1 file changed, 1 insertion(+)

diff --git a/crossenv/scripts/site.py.tmpl b/crossenv/scripts/site.py.tmpl
index 14ac777..63e4d6e 100644
--- a/crossenv/scripts/site.py.tmpl
+++ b/crossenv/scripts/site.py.tmpl
@@ -106,6 +106,7 @@ class CrossenvFinder(importlib.abc.MetaPathFinder):
         'os': '{{context.lib_path}}/os-patch.py',
         'sysconfig': '{{context.lib_path}}/sysconfig-patch.py',
         'distutils.sysconfig': '{{context.lib_path}}/distutils-sysconfig-patch.py',
+        'distutils.sysconfig_pypy': '{{context.lib_path}}/distutils-sysconfig-patch.py',
         'platform': '{{context.lib_path}}/platform-patch.py',
         'pkg_resources': '{{context.lib_path}}/pkg_resources-patch.py',
         'pip._vendor.pkg_resources': '{{context.lib_path}}/pkg_resources-patch.py',
